src/bookList-detail.js

```diff
@@ -0,0 +1,81 @@
+
+import { LitElement, html} from 'lit-element';
+import './search-bar.js';
+import styles from './styles/bookslistStyles.js';
+
+
+class BookListDetail extends LitElement {
+
+    static get properties() {
+        return {
+            repo: { type: Array },
+            generalrepo: { type: Array },
+            myArray: { type: Array },
+        };
+    }
+
+    static get styles(){
+        return [styles];
+    }
+
+    constructor() {
+        super();
+        this.repo = [];
+        this.generalrepo = [];
+        this.myArray = [];
+        this.getDataJSON();
+        this.datafromSerchBar();
+    }
+
+    render() {
+        return html`
+          ${this.dataTemplate} 
+        `;
+    }
+    
+    async getDataJSON() {
+        const datos = await fetch('../infoBooks.json').then(response => response.json());
+        this.generalrepo = datos;
+        this._dataFormat();
+    }
+    
+    datafromSerchBar(){
+        addEventListener('Apidata',(e)=>
+        this.myArray =e.detail.data
+        );
+    }
+
+    _dataFormat() {
+        let characters = this.generalrepo.map(function(element){
+            return [element.cover, element.autorb, element.titleb];
+        });
```

Comment:
I recommend you to create an object
```suggestion
        let characters = this.generalrepo.map(({cover, autorb, titleb}) => {
            return {cover, autorb, titleb};
        });
```

Reply:
thanks for you recommendation. This change was applied!!